contract-ui/tabs/explorer/page.tsx

```diff
@@ -0,0 +1,37 @@
+import { useContractFunctions } from "../../../components/contract-components/hooks";
+import { ImportContract } from "../../../components/contract-components/import-contract";
+import { ContractFunctionsOverview } from "../../../components/contract-functions/contract-functions";
+import { Flex, Skeleton } from "@chakra-ui/react";
+import { useContract } from "@thirdweb-dev/react";
+import { Abi } from "@thirdweb-dev/sdk";
+
+interface ContractCodePageProps {
+  contractAddress?: string;
+}
+
+export const ContractExplorerPage: React.FC<ContractCodePageProps> = ({
+  contractAddress,
+}) => {
+  const { contract, isSuccess, isError } = useContract(contractAddress);
+
+  const functions = useContractFunctions(contract?.abi as Abi);
+  if (!contractAddress) {
+    return <div>No contract address provided</div>;
+  }
+
+  if ((!contract?.abi && isSuccess) || isError) {
```

Comment:
do we need to do this here? right now we only do it on overview

Reply:
yeah only overview is fine for this